test/framework/gardener_utils.go

```diff
@@ -53,13 +56,46 @@ func (f *GardenerFramework) GetSeed(ctx context.Context, seedName string) (*gard
 		return nil, nil, fmt.Errorf("could not get Seed from Shoot in Garden cluster: %w", err)
 	}
 
-	seedSecretRef := seed.Spec.SecretRef
-	if seedSecretRef == nil {
-		f.Logger.Info("Seed does not have secretRef set, skip constructing seed client")
-		return seed, nil, nil
+	managedSeed := &seedmanagementv1alpha1.ManagedSeed{}
+	if err := f.GardenClient.Client().Get(ctx, kubernetesutils.Key(v1beta1constants.GardenNamespace, seed.Name), managedSeed); err != nil {
+		if apierrors.IsNotFound(err) {
+			f.Logger.Info("Seed is not a ManagedSeed, checking seed.spec.secretRef")
+
+			seedSecretRef := seed.Spec.SecretRef
```

Comment:
will the `seed.spec.secretRef` field be deprecated and removed completely or will it continue to exist for `seeds` that are no `managedSeeds`

otherwise, lgtm

Reply:
> will the seed.spec.secretRef field be deprecated

Yes, that's the plan.

Reply:
Ok, thanks for the clarification. 

Then in future, it won't be possible anymore to run certain tests like https://github.com/gardener/gardener/blob/master/.test-defs/HibernateShoot.yaml with `seeds` that are not `managedSeeds`.

Reply:
Yes, we have to find a way to workaround that.